### PR TITLE
fix: Allow refs on lazy components

### PR DESCRIPTION
**Pull request by eps1lon**
See facebook/react#15318 for full context

### CODE REVIEW

File: test/AppContainer.dev.test.js [hunk 1]
```diff
@@ -927,6 +928,7 @@ describe(`AppContainer (dev)`, () => {
       await Promise.resolve(1)
 
       expect(spy).not.toHaveBeenCalled()
+      expect(ref).toHaveBeenCalledTimes(1)
```
Comment (eps1lon) [inline test/AppContainer.dev.test.js]: When running only this test it worked but when running all it fails. Those things should probably be wrapped in `act` to work properly. I don't have enough context here to make a good judgement how the full test should work with regards to timing.

File: src/reconciler/fiberUpdater.js [hunk 2]
```diff
@@ -18,12 +18,21 @@ export const updateLazy = (target, type) => {
         const C = resolveType(m.default)
         // chunks has been updated - new hot loader process is taking a place
         enterHotUpdate()
+        if (!React.forwardRef) {
```
Comment (eps1lon) [inline src/reconciler/fiberUpdater.js]: I thought this path was only used for `React.lazy` which is a 16.6 feature where React.forwardRef should be available. But react 15 uses this path as well which caused https://travis-ci.org/gaearon/react-hot-loader/jobs/516138894#L790

**Comment by theKashey** [comment]
Excellent! Thank you good context and a quick fix.

**Comment by theKashey** [comment]
Released

**Pull request merged**